python/triton/runtime/autotuner.py

```diff
@@ -14,26 +16,28 @@
 class Autotuner(KernelInterface):
 
     def __init__(
-        self,
-        fn,
-        arg_names,
-        configs,
-        key,
-        reset_to_zero,
-        restore_value,
-        pre_hook=None,
-        post_hook=None,
-        prune_configs_by: Optional[Dict] = None,
-        warmup=None,
-        rep=None,
-        use_cuda_graph=False,
-        do_bench=None,
+            self,
+            fn,
+            arg_names,
+            configs,
+            key,
+            reset_to_zero,
+            restore_value,
+            pre_hook=None,
+            post_hook=None,
+            prune_configs_by: Optional[Dict] = None,
+            warmup=None,
+            rep=None,
+            use_cuda_graph=False,
+            do_bench=None,
+            max_workers: int = os.cpu_count(),
```

Comment:
If this PR is going to be merged, I would prefer the default value to be `1` for backward compatibility and resource saving. Some machines I only have a small DRAM

Reply:
I was considering introducing a new argument or environment variable to enable parallel compilation, which would set max_workers to os.cpu_count(). Otherwise, it would default to 1. 

What would be your recommendation for the best approach to implement this?